core/src/test/java/kafka/test/ClusterConfig.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.test;
+
+import kafka.test.annotation.Type;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+
+import java.io.File;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * Represents a requested configuration of a Kafka cluster for integration testing
+ */
+public class ClusterConfig {
+
+    private final Type type;
+    private final int brokers;
+    private final int controllers;
+    private final String name;
+    private final boolean autoStart;
+
+    private final String securityProtocol;
+    private final String listenerName;
+    private final File trustStoreFile;
+
+    private final Properties serverProperties = new Properties();
+    private final Properties producerProperties = new Properties();
+    private final Properties consumerProperties = new Properties();
+    private final Properties adminClientProperties = new Properties();
+    private final Properties saslServerProperties = new Properties();
+    private final Properties saslClientProperties = new Properties();
+
+    ClusterConfig(Type type, int brokers, int controllers, String name, boolean autoStart,
+                  String securityProtocol, String listenerName, File trustStoreFile) {
+        this.type = type;
+        this.brokers = brokers;
+        this.controllers = controllers;
+        this.name = name;
+        this.autoStart = autoStart;
+        this.securityProtocol = securityProtocol;
+        this.listenerName = listenerName;
+        this.trustStoreFile = trustStoreFile;
+    }
+
+    public Type clusterType() {
+        return type;
+    }
+
+    public int brokers() {
```

Comment:
let's use "numBrokers" , etc. to reflect the fact that we're not returning the actual brokers.  Same for "controllers"